gemonlinedb/src/common/GEMOnlineDBManagerWeb.cc

```diff
@@ -65,9 +152,87 @@ void gem::onlinedb::GEMOnlineDBManagerWeb::applicationPage(xgi::Input* in, xgi::
 void gem::onlinedb::GEMOnlineDBManagerWeb::jsonUpdate(xgi::Input* in, xgi::Output* out)
   throw (xgi::exception::Exception)
 {
+  // We use jsonUpdate for the basic statistics that are always visible in the
+  // first tab. Additional information is retrieved from other sources.
+
   CMSGEMOS_DEBUG("GEMOnlineDBManagerWeb::jsonUpdate");
   out->getHTTPResponseHeader().addHeader("Content-Type", "application/json");
-  *out << " { " << std::endl;
-  // fill JSON if needed
-  *out << " } " << std::endl;
+
+  json_auto_t *answer = janssonCheckPtr(json_object());
```

Comment:
Jansson objects are reference-counted, so standard C++ smart pointers won't work (notice there's no call to `free()`). `json_auto_t` is a smart pointer that decrements the reference count when going out of scope (using `__attribute__((cleanup(function)))`).

Reply:
OK, I was mostly concerned because the last line I mention explicitly calls `free` on the `json` object, so I was concerned.
Maybe my comment should only apply to L235-237.
If you're not leaking, then I guess I won't object.